### PR TITLE
quieten down docker

### PR DESCRIPTION
**Pull request by mixmix**
Our tests are pretty spammy with docker spinning up / down. 

This makes it a lot easier to see where "actual" tests are up to, and also for us to see that there is still logging coming out of substrate.

image

### CODE REVIEW

File: dev/bin/spin-up.sh [hunk 1]
```diff
@@ -5,5 +5,5 @@ docker_file=dev/docker-scripts/$1.yaml
 if [ $GITHUB_WORKSPACE ]; then
   docker compose --file $docker_file up --detach --quiet-pull;
 else
-  docker compose --file $docker_file up --detach;
+  docker compose --file $docker_file --progress quiet up --detach;
```
Comment (mixmix) [inline dev/bin/spin-up.sh:8]: We might be able to make it output only errors?

e.g. would this work?

```bash
#!/bin/bash

# Run docker compose up, suppressing stdout and stderr
docker compose up -d > /dev/null 2>&1

# Check if the command failed
if [ $? -ne 0 ]; then
    echo "Error: 'docker compose up' failed. See details below:"
    # Run the command again without suppressing output to display the error
    docker compose up
fi
```

Comment (mixmix) [reply dev/bin/spin-up.sh]: Ended up doing something like this

File: dev/bin/remove-db.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #! /usr/bin/env bash
 
 rm -rf .entropy
-docker compose --file dev/docker-scripts/four-nodes.yaml down 2> /dev/null
-docker compose --file dev/docker-scripts/one-node-no-tss.yaml down 2> /dev/null
+./dev/bin/spin-down.sh four-nodes
+./dev/bin/spin-down.sh one-node-no-tss
```
Comment (mixmix) [inline dev/bin/remove-db.sh:5]: this works, but it adds a call `docker compose --file "$FILE" logs` :shrug:

File: dev/bin/spin-up.sh [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
+
+spin_up() {
+  docker compose --file "$FILE" "$@" up --detach
+
+  if [ $? -ne 0 ]; then
+    echo "Error: 'docker compose up' failed."
+    echo "Retrying verbose:"
+    docker compose --file "$FILE" up --detach
+  fi
+}
```
Comment (mixmix) [inline dev/bin/spin-up.sh:14]: Tested this and here's how it looked when if fails (make it fail by editing `ENTROPY_CORE_VERSION`)

image

Comment (mixmix) [reply dev/bin/spin-up.sh:14]: This assumes docker is good and exits with an exit code if it hit any problems...

**Comment by mixmix** [comment]
@frankiebee can review at leisure

**Comment by mixmix** [comment]
Frankie is gonna test how this behaves in CI

**Review by rh0delta — APPROVED** [review]
this looks good to me, but you my still want @frankiebee's sign off too

**Comment by frankiebee** [comment]
note: i'll get a screen shot of ci failing in a different branch then i'll approve and merge if i like it

**Comment by mixmix** [comment]
@frankiebee here's the branch where you see it failing: https://github.com/entropyxyz/sdk/pull/431

**Review by frankiebee — APPROVED** [review]

**Pull request merged**